lib/thread.c

```diff
@@ -61,7 +62,9 @@ static struct list *masters;
 /* CLI start ---------------------------------------------------------------- */
 static unsigned int cpu_record_hash_key(struct cpu_thread_history *a)
 {
-	return (uintptr_t)a->func;
+	int size = sizeof (a->func);
+
+	return jhash(a->func, size, 0);
```

Comment:
You're missing a & on `&a->func` here and are now hashing the actual binary code of the function :D

Reply:
I'll fix this.